### PR TITLE
feat(Core): migrate engine extensions to rust core

### PR DESCRIPTION
**Pull request by ShadowApex**

### CODE REVIEW

File: core/global/launch_manager.gd [hunk 1]
```diff
@@ -59,7 +59,7 @@ var _persist_path: String = "/".join([_data_dir, "launcher.json"])
 var _persist_data: Dictionary = {"version": 1}
 var _ogui_window_id := 0
 var should_manage_overlay := true
-var logger := Log.get_logger("LaunchManager", Log.LEVEL.INFO)
```
Comment (pastaq) [inline core/global/launch_manager.gd:60]: Note for later, fixme

File: core/systems/launcher/running_app.gd [hunk 2]
```diff
@@ -102,7 +102,7 @@ var steam_close_tries := 0
 ## Flag for if OGUI should manage this app. Set to false if app is launched
 ## outside OGUI and we just want to track it.
 var is_ogui_managed: bool = true
-var logger := Log.get_logger("RunningApp", Log.LEVEL.INFO)
+var logger := Log.get_logger("RunningApp", Log.LEVEL.TRACE)
```
Comment (pastaq) [inline core/systems/launcher/running_app.gd:105]: Note for later, fixme

File: core/systems/launcher/running_app.gd [hunk 3]
```diff
@@ -111,6 +111,7 @@ func _init(item: LibraryLaunchItem, process_id: int, dsp: String) -> void:
 	display = dsp
 
 
+# TODO: Only call this on window creation/deletion
```
Comment (pastaq) [inline core/systems/launcher/running_app.gd:114]: When is it running now?

File: core/global/launch_manager.gd [hunk 4]
```diff
@@ -108,9 +108,21 @@ func _init() -> void:
...
 		_xwayland_primary.focused_app_updated.connect(on_focused_app_changed)
 
+		# Listen for when focusable apps change
+		var on_focusable_apps_changed := func(from: PackedInt64Array, to: PackedInt64Array):
+			if from == to:
+				return
+			logger.debug("Focusable apps changed from", from, "to", to)
+			# If focusable apps has changed and the currently focused app no longer exists,
+			# remove the manual focus
+			var baselayer_app := _xwayland_primary.baselayer_app
```
Comment (pastaq) [inline core/global/launch_manager.gd:120]: What is a base layer app and how is it defined?

File: extensions/Cargo.lock [hunk 5]
```diff
@@ -522,7 +522,7 @@ dependencies = [
 [[package]]
 name = "gamescope-x11-client"
 version = "0.1.0"
-source = "git+https://github.com/ShadowBlip/gamescope-x11-client?branch=main#3a0cbe64ba60dffb5ad85f156101c056f764e659"
+source = "git+https://github.com/ShadowBlip/gamescope-x11-client?branch=main#deeab5be067bfbb2add2446d8c7fbcbeba7c8c7f"
```
Comment (pastaq) [inline extensions/Cargo.lock:525]: Why did you commit this?

Comment (ShadowApex) [reply extensions/Cargo.lock:525]: To update the version of the `gamescope-x11-client` crate.

File: core/global/launch_manager.gd (hunk 4 above)
Comment (ShadowApex) [reply core/global/launch_manager.gd:120]: The baselayer app is a gamescope property (`GAMESCOPECTRL_BASELAYER_APP`) that can be set with an app id to manually make gamescope focus a specific app.

File: core/systems/launcher/running_app.gd (hunk 3 above)
Comment (ShadowApex) [reply core/systems/launcher/running_app.gd:144]: Every frame

File: extensions/Cargo.lock (hunk 5 above)
Comment (pastaq) [reply extensions/Cargo.lock:525]: Isn't that done in the toml and the lock gets updated and committed during CI?

**Comment by philmmanjaro** [comment]
I briefly tested `2040d5e`. This potentially fixes https://github.com/ShadowBlip/OpenGamepadUI/issues/397. However, it seems I don't see the TDP functions yet in overlay nor full-session mode. Will test some more on the other functions given.

**Comment by philmmanjaro** [comment]
When I tried `708e9ee` and clicked on `Power` on the left menu OGUI somehow restarted itself. The power menu never popped up. So restart, sleep, changing to desktop mode or shutdown is currently not working. Other things are fixed and a lot got improved. Great job so far.

File: core/global/launch_manager.gd (hunk 1 above)
Comment (ShadowApex) [reply core/global/launch_manager.gd:60]: ok

File: core/systems/input/overlay_mode_input_manager.gd [hunk 6]
```diff
@@ -71,7 +75,19 @@ func _send_input(dbus_path: String, action: String, pressed: bool, strength: flo
...
+	if input_plumber.is_running() and not InputManager.is_inputplumber_event(event):
+		if event is InputEventJoypadButton or event is InputEventJoypadMotion:
+			logger.debug("Skipping Godot event while InputPlumber is running:", event)
+			get_viewport().set_input_as_handled()
+			return
+
 	var dbus_path := event.get_meta("dbus_path", "") as String
+	
+	# Don't process Godot events if InputPlumber is running
+	if input_plumber.is_running() and not InputManager.is_inputplumber_event(event):
```
Comment (ShadowApex) [inline core/systems/input/overlay_mode_input_manager.gd:88]: Handle this the same way as full input manager

**Review by pastaq — APPROVED** [review]
LGTM

**Comment by philmmanjaro** [comment]
Also tested it on my end. Works better than the version we have currently in `main` branch.

**Pull request merged**